src/Analysis/Codelyzer.Analysis.Build/WorkspaceBuilderHelper.cs

```diff
@@ -578,41 +574,31 @@ public static string GetFrameworkMsBuildExePath(bool projectUsesSdk = false,stri
             var programFiles = programFilesPath ?? Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
             string programFilesX86 = programFilesX86Path ?? System.Environment.GetFolderPath(System.Environment.SpecialFolder.ProgramFilesX86);
 
-
-            if (projectUsesSdk)
+           
+            //2022              
+            vsDirectory = new DirectoryInfo(Path.Combine(programFiles, "Microsoft Visual Studio"));
+            msbuildpath = GetMsBuildPathFromVSDirectory(vsDirectory, editions, targets);
+            if (!string.IsNullOrEmpty(msbuildpath))
```

Comment:
Why did u remove the UseSdk check? Not sure I understand if this logic is still being applied.

Reply:
This SDK check was a workaround for a workaround. Now that we are targeting .NET 6 and updated buildalyzer, we can revert back to the original MSBuild hierarchy we had (and thus no longer need special handling for building SDK projects).

Reply:
As a refresher, we used the `projectUsesSdk` flag so we could still build .NET 6 projects. Now that we are using a later version of Buildalyzer and Codelyzer targets .NET 6, we no longer require this special case. We are free to prioritize the latest version of MSBuild.